src/poetry/core/masonry/builders/sdist.py

```diff
@@ -55,10 +56,18 @@ class SdistBuilder(Builder):
 
     format = "sdist"
 
-    def build(self, target_dir: Path | None = None) -> Path:
+    def __init__(
+        self,
+        poetry: Poetry,
+        executable: Path | None = None,
+        target_dir: Path | None = None,
+    ) -> None:
+        super().__init__(poetry, executable=executable)
+        self._target_dir = target_dir
```

Comment:
Not sure if this is desired. Iirc build is intended to be called multiple times with different target directories. You can default to cwd if desired though.

Reply:
The builders were inconsistent about whether they took a target directory on their constructor or on their `build()` method.  I have made them consistently put it on the constructor.

I think the suggestion is that we should jump the other way?  ie remove the parameter from the constructors, put it on the `build()` method?

(There's a default value either way)

Reply:
I suspect the inconsistency comes from the use of `make_in` in the wheel builder. Happy for it to be unified as consistency here is better, with the following funcitonality maintained.

1. There is a default target directory (which as you said is present either way).
2. the `build()` method can accept target directory.

This can also mean that make_in / make becomes redundant. Will need to check downstream see if they are being used.

Reply:
I pushed a commit that does what I think we're agreeing - say if I've misunderstood

Reply:
`make()` and `make_in()` do indeed look redundant, happy to remove them if you want